clarity/src/vm/database/clarity_db.rs

```diff
@@ -700,6 +687,10 @@ impl<'a> ClarityDatabase<'a> {
         Ok(())
     }
 
+    pub fn get_cc_special_cases_handler(&self) -> Option<SpecialCaseHandler> {
+        self.store.get_cc_special_cases_handler()
+    }
+
```

Comment:
I'm confused here -- why is `insert_microblock_poison` preserved in the clarity DB, but the top-level method for processing it in an `OwnedEnvironment` is dropped?

Reply:
`insert_microblock_poison` should probably also be moved out of this module.

The reason for removing the method for processing microblock poison is that it is a method that depends on the `StacksChainState` struct, and the Clarity crate cannot depend on anything in the main source tree.

Reply:
Thanks, makes sense to me.